core/src/main/scala/org/genivi/sota/core/VehicleUpdatesResource.scala

```diff
@@ -154,28 +155,34 @@ class VehicleUpdatesResource(db : Database, resolverClient: ExternalResolverClie
   /**
     * The web app PUT the status of the given ([[UpdateSpec]], VIN) to [[UpdateStatus.Canceled]]
     */
-  def cancelUpdate(vin: Vehicle.Vin, uuid: Refined[String, Uuid]): Route = {
-    val response = db.run(UpdateSpecs.cancelUpdate(vin, uuid)).map {
-      i: Int => i match {
-          case 0 => HttpResponse(StatusCodes.BadRequest)
-          case _ => HttpResponse(StatusCodes.NoContent)
-        }
+  def cancelUpdate(ns: Namespace, vin: Vehicle.Vin, uuid: Refined[String, Uuid]): Route = {
+    val response = db.run(UpdateSpecs.cancelUpdate(ns, vin, uuid)).map {
+      case 0 => HttpResponse(StatusCodes.BadRequest)
+      case _ => HttpResponse(StatusCodes.NoContent)
     }
     complete(response)
   }
 
   val route = {
     (pathPrefix("api" / "v1" / "vehicle_updates") & extractVin) { vin =>
-      (put & path("installed")) { updateInstalledPackages(vin) } ~
-      (put & path("order")) { setInstallOrder(vin) } ~
-      (put & extractUuid & path("cancelupdate") ) { uuid => cancelUpdate(vin, uuid) } ~
-      (post & namespaceExtractor & pathEnd) { ns => queueVehicleUpdate(ns, vin) } ~
-      (get & pathEnd) { logVehicleSeen(vin) { pendingPackages(vin) } } ~
-      (get & path("queued")) { pendingPackages(vin) } ~
-      (get & extractUuid & path("download")) { uuid => downloadPackage(vin, uuid) } ~
-      (get & path("operationresults")) { results(vin) } ~
-      (post & extractUuid) { reportInstall } ~
-      (post & path("sync")) { sync(vin) }
+      get {
+        pathEnd { logVehicleSeen(vin) { pendingPackages(vin) } } ~
+        path("queued") { pendingPackages(vin) } ~
+        path("operationresults") { results(vin) } ~
+        (extractUuid & path("download")) { uuid => downloadPackage(vin, uuid) }
+      } ~
+      put {
+        path("installed") { updateInstalledPackages(vin) } ~
+        path("order") { setInstallOrder(vin) } ~
+        (extractUuid & path("cancelupdate") ) { uuid =>
```

Comment:
Maybe this should be a `DELETE` request? Or a PUT/PATCH with a different resource representation, with the status changed, Rather than a specific path to change a resource.